sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -106,21 +139,39 @@ public DigitalTwinsAsyncClient buildAsyncClient() {
         Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
         Objects.requireNonNull(endpoint, "'endpoint' cannot be null");
 
+        Configuration buildConfiguration = this.configuration;
+        if (buildConfiguration == null)
+        {
+            buildConfiguration = Configuration.getGlobalConfiguration().clone();
+        }
```

Comment:
In the configurations client code, their builder never overwrites the values of the instance variables in the builder. Instead it creates local copies that can be altered instead

Reply:
Do we know why? Is this more efficient, or is it the defined process?